gst/nnstreamer/tensor_query/tensor_query_serversrc.c

```diff
@@ -189,6 +190,152 @@ gst_tensor_query_serversrc_finalize (GObject * object)
   G_OBJECT_CLASS (parent_class)->finalize (object);
 }
 
+/**
+ * @brief nnstreamer-edge event callback.
+ */
+static int
+_nns_edge_event_cb (nns_edge_event_h event_h, void *user_data)
+{
+  nns_edge_event_e event_type;
+  int ret = NNS_EDGE_ERROR_NONE;
+
+  GstTensorQueryServerSrc *src = (GstTensorQueryServerSrc *) user_data;
+  if (0 != nns_edge_event_get_type (event_h, &event_type)) {
+    nns_loge ("Failed to get event type!");
+    return NNS_EDGE_ERROR_UNKNOWN;
+  }
+
+  switch (event_type) {
+    case NNS_EDGE_EVENT_NEW_DATA_RECEIVED:
+    {
+      nns_edge_data_h data;
+
+      nns_edge_event_parse_new_data (event_h, &data);
+      g_async_queue_push (src->msg_queue, data);
+      break;
+    }
+    default:
+      break;
+  }
+
+  return ret;
+}
+
+/**
+ * @brief start processing of query_serversrc, setting up the server
+ */
+static gboolean
+_gst_tensor_query_serversrc_start (GstTensorQueryServerSrc * src)
+{
+  g_autofree gchar *id_str = NULL;
+
+  id_str = g_strdup_printf ("%d", src->src_id);
+  src->server_h = gst_tensor_query_server_add_data (id_str);
+
+  if (!gst_tensor_query_server_wait_sink (src->server_h)) {
+    nns_loge ("Failed to get server information from query server.");
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief start processing of query_serversrc, setting up the server
+ */
+static gboolean
+_gst_tensor_query_serversrc_playing (GstTensorQueryServerSrc * src,
+    nns_edge_connect_type_e connect_type)
+{
+  g_autofree char *id_str = NULL, *port = NULL;
+  gboolean ret = TRUE;
+
+  id_str = g_strdup_printf ("%d", src->src_id);
+  src->edge_h = gst_tensor_query_server_get_edge_handle (id_str, connect_type);
+
+  if (!src->edge_h) {
+    return FALSE;
+  }
+
+  if (src->host)
+    nns_edge_set_info (src->edge_h, "HOST", src->host);
+  if (src->port > 0) {
+    port = g_strdup_printf ("%u", src->port);
+    nns_edge_set_info (src->edge_h, "PORT", port);
+  }
+  if (src->dest_host)
+    nns_edge_set_info (src->edge_h, "DEST_HOST", src->dest_host);
+  if (src->dest_port > 0) {
+    port = g_strdup_printf ("%u", src->dest_port);
```

Comment:
line 263 and 269 ptr 'port' gets different ptr for host/dest port string.

Reply:
Updated.